src/Arcus.BackgroundJobs.Tests.Integration/Fixture/ServiceBus/TestSpyRestartServiceBusMessagePump.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Arcus.Messaging.Abstractions.ServiceBus.MessageHandling;
+using Arcus.Messaging.Pumps.ServiceBus;
+using Arcus.Messaging.Pumps.ServiceBus.Configuration;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Moq;
+
+namespace Arcus.BackgroundJobs.Tests.Integration.Fixture.ServiceBus
+{
+    /// <summary>
+    /// Represents an <see cref="AzureServiceBusMessagePump"/> implementation that spies on whether the message pump was triggered for restart.
+    /// </summary>
+    internal class TestSpyRestartServiceBusMessagePump : AzureServiceBusMessagePump
```

Comment:
Created a dedicated spy test fixture to simplify the test that verifies that we can auto-restart message pumps based on Key vault events. It removes the need for a fully working message pump, handling, and EventGrid connection; as all those things are in fact not in the scope of this feature (tested in Messaging). The test only needs to know that the message pump was triggered for restart, not that the pump is still valid after restart (tested in Messaging).